Doc/library/os.path.rst

```diff
@@ -304,12 +304,13 @@ the :mod:`glob` module.)
 
 .. function:: join(path, *paths)
 
-   Join one or more path components intelligently.  The return value is the
-   concatenation of *path* and any members of *\*paths* with exactly one
-   directory separator following each non-empty part except the last, meaning
-   that the result will only end in a separator if the last part is empty.  If
-   a component is an absolute path, all previous components are thrown away
-   and joining continues from the absolute path component.
+   Join one or more path components intelligently. The return value is the
+   concatenation of path and any members of *\*paths* so that there is a
+   directory separator (os.sep) following each part except the last.
+   An empty part is ignored unless it is the last part, in which case the result
+   will end in a separator.  If a component is an absolute path, all previous
```

Comment:
In looking into this, it looks like I'm actually the one who last made major changes to this section 7 years ago, and I now agree that I did leave it rather confusing.  Oops :)

I think the following is a slightly better formulation; do you agree?

```suggestion
   concatenation of *path* and any non-empty members of *\*paths* separated
   by the platform's directory separator character.  If the last member of *\*paths*
   is empty, the result will end in a directory separator.
   If a component is an absolute path, all previous
```